torch/_weights_only_unpickler.py

```diff
@@ -162,10 +283,28 @@ def load(self):
             elif key[0] == REDUCE[0]:
                 args = self.stack.pop()
                 func = self.stack[-1]
-                if func not in _get_allowed_globals().values():
+                if (
+                    func not in _get_allowed_globals().values()
+                    and func not in _get_user_allowed_globals().values()
+                ):
                     raise RuntimeError(
                         f"Trying to call reduce for unrecognized function {func}"
                     )
+                # Prevent tensor subclass type that is pushed onto stack in GLOBAL from
```

Comment:
We do not want to expect new/init are not changed. So we cannot allow them to be called.

Reply:
To elaborate here, we want to make sure that tensor subclass types that have been allowed by the Unpickler but are not in allowed globals are **only ever used as the second argument to `torch.Tensor._rebuild_from_type_v2`** 

We want to disallow the tensor subclass type from being pushed onto the stack as an argument for any `REBUILD` instruction, in case one of its arguments (e.g. as `func` for line 58 here) 

https://github.com/pytorch/pytorch/blob/4e66aaa01092ddc8822bbca315b673329c76f4cd/torch/_tensor.py#L57-L71

If this happens, `__init__` and `__new__` of the tensor subclass might be called as mentioned above.